mypyc/irbuild/ll_builder.py

```diff
@@ -1079,12 +1081,15 @@ def int_op(self, type: RType, lhs: Value, rhs: Value, op: int, line: int) -> Val
     def comparison_op(self, lhs: Value, rhs: Value, op: int, line: int) -> Value:
         return self.add(ComparisonOp(lhs, rhs, op, line))
 
-    def builtin_len(self, val: Value, line: int) -> Value:
+    def builtin_len(self, val: Value, line: int,
+                    get_c_pyssize_t_rprimitive: bool = False) -> Value:
         typ = val.type
         if is_list_rprimitive(typ) or is_tuple_rprimitive(typ):
             elem_address = self.add(GetElementPtr(val, PyVarObject, 'ob_size'))
             size_value = self.add(LoadMem(c_pyssize_t_rprimitive, elem_address))
             self.add(KeepAlive([val]))
+            if get_c_pyssize_t_rprimitive:
+                return size_value
```

Comment:
@JukkaL Is it OK that I directly get `c_pyssize_t_rprimitive` from here?

Reply:
This is fine, as long as it's supported for all types below. Otherwise it would be error-prone.

Reply:
Got it! I add support for `dict` and `set` below

Reply:
The generic case also needs to return a `c_pyssize_t_rprimitive`. For this you can call `PyObject_Size` directly instead of `CPyObject_Size` (via `generic_len_op`). Add a new primitive to `mypyc.primitives.generic_ops`, e.g. `generic_ssize_t_len_op` or similar.

Reply:
Thanks for pointing out! I will fix it later.